src/server/game/Server/WorldSocket.cpp

```diff
@@ -161,43 +161,46 @@ void WorldSocket::CheckIpCallback(PreparedQueryResult result)
 bool WorldSocket::Update()
 {
     EncryptableAndCompressiblePacket* queued;
-    MessageBuffer buffer(_sendBufferSize);
-    while (_bufferQueue.Dequeue(queued))
+    if (_bufferQueue.Dequeue(queued))
     {
-        queued->CompressIfNeeded();
+        // Allocate buffer only when it's needed but not on every Update() call.
+        MessageBuffer buffer(_sendBufferSize);
+        do
+        {
+            queued->CompressIfNeeded();
+            ServerPktHeader header(queued->size() + 2, queued->GetOpcode());
+            if (queued->NeedsEncryption())
+                _authCrypt.EncryptSend(header.header, header.getHeaderLength());
 
-        ServerPktHeader header(queued->size() + 2, queued->GetOpcode());
-        if (queued->NeedsEncryption())
-            _authCrypt.EncryptSend(header.header, header.getHeaderLength());
+            if (buffer.GetRemainingSpace() < queued->size() + header.getHeaderLength())
+            {
+                QueuePacket(std::move(buffer));
```

Comment:
First time reviewing code here but out of curiosity, is there a need to queue an empty buffer?

IIUC right now the buffer doesn't have any information and we're queueing an empty packet?

This part of the code seems like a no-op.

Reply:
Ahh this is done in a do loop. False alarm!